tests/utils/parseUrl.test.js

```diff
@@ -0,0 +1,84 @@
+require = require('esm')(module);
+
+const parseUrlFile = require('../../utils/parseUrl');
+
+const parseUrl = parseUrlFile.parseUrl;
+
+describe('parseUrl', () => {
+  it('parses non beta package', () => {
+    expect(parseUrl('https://unpkg.com/pkg-dash@1.0.0/extra.js')).toEqual({
+      name: 'pkg-dash',
+      version: '1.0.0',
+      path: '/pkg-dash@1.0.0/extra.js',
```

Comment:
I always considered the path as the full path minus the package name and the file name. What is everyone else's thoughts on this definition? @alex-saunders @VirtualDOMinic

Reply:
I was not sure of this definition, I assumed it meant the unpkg url. 

I assume you mean `https://unpkg.com/@scope/pkg@1.0.0/es6/index.js` file would be: `index.js` and path would be `/es6/index.js`?

Reply:
I'd be inclined to agree with @lukejacksonn (and yep @kiraarghy my understanding of it matches yours -- i.e. the current `path` would be the equivalent of `${name}@${version}/${newPath}` where `newPath` is the `path` Luke is suggesting

Reply:
I think we used to use it for telling what folder the resource was in.. so:
```
https://unpkg.com/@scope/pkg@1.0.0/es6/index.js
```
then:
```js
{
  name: '@scope/pkg',
  version: '1.0.0',
  path: 'es6',
  file: 'index.js'
}
```

Then you could reconstruct the URL in full just by combining all the parts. Maybe we could store the full path (like you do here) then create a new key called `folder` or something?

Reply:
that's cool with me 😄 requirements are hard.

Reply:
They are, but you are doing great 🙌

Reply:
Okay so final structure is such: 

If I called it with: `https://unpkg.com/@scope/pkg@1.0.0/es6/index.js`

```javascript 
{
  name: '@scope/pkg',
  version: '1.0.0',
  directory: 'es6',
  path: /@scope/pkg@1.0.0/es6/index.js
  file: 'index.js'
}
```